### PR TITLE
Fix mpu part size round off error for large files

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- Fixes a rounding error for large files.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/s3_util_tests.c [hunk 1]
```diff
@@ -364,6 +364,120 @@ static int s_test_s3_get_num_parts_and_get_part_range(struct aws_allocator *allo
...
+        uint32_t num_parts;
+
+        ASSERT_SUCCESS(aws_s3_calculate_optimal_mpu_part_size_and_num_parts(
+            content_length,
+            valid_request_part_config[i].client_part_size,
+            valid_request_part_config[i].client_max_part_size,
+            &part_size,
+            &num_parts));
+
+        ASSERT_TRUE(num_parts <= g_s3_max_num_upload_parts);
```
Comment (DmitriyMusatkin) [inline tests/s3_util_tests.c:456]: should we lock down exact number of parts and part size expected to avoid any unexpected changes in future?

Comment (waahm7) [reply tests/s3_util_tests.c]: Thanks, updated.

File: tests/s3_util_tests.c [hunk 2]
```diff
@@ -364,6 +364,151 @@ static int s_test_s3_get_num_parts_and_get_part_range(struct aws_allocator *allo
...
+        {
+            .name = "bump content length with non-zero mod",
+            .content_length = 999999,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 100,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "5 tb content length",
```
Comment (TingDaoK) [inline tests/s3_util_tests.c:465]: the numbers in real life is based on Bytes, instead of MB.

**Review by TingDaoK — APPROVED** [review]

File: tests/s3_util_tests.c [hunk 3]
```diff
@@ -364,6 +364,151 @@ static int s_test_s3_get_num_parts_and_get_part_range(struct aws_allocator *allo
...
+            .content_length = 5 * 1024 * 1024,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 525,
+            .expected_num_parts = 9987,
+        },
+    };
+    for (size_t i = 0; i < AWS_ARRAY_SIZE(valid_request_part_config); ++i) {
+        printf("valid example [%zu]: %s\n", i, valid_request_part_config[i].name);
```
Comment (DmitriyMusatkin) [inline tests/s3_util_tests.c:474]: nit: logf_info might be a better option here

**Review by DmitriyMusatkin — APPROVED** [review]
fix and ship

File: tests/s3_util_tests.c (hunk 2 above)
Comment (waahm7) [reply tests/s3_util_tests.c:466]: Thanks, fixed.

**Pull request merged**